core/src/com/unciv/logic/city/PopulationManager.kt

```diff
@@ -109,34 +111,63 @@ class PopulationManager {
         addPopulation(-population + count)
     }
 
-    internal fun autoAssignPopulation(foodWeight: Float = 1f) {
+    internal fun autoAssignPopulation() {
+        cityInfo.cityStats.update()  // calculate current stats with current assignments
+        // save temporaries
+        cityInfo.currentSurplusFood = cityInfo.foodForNextTurn().toFloat()  // save temporary for calculations
+        cityInfo.currentGPPBonus = 0
+        for (unique in cityInfo.getMatchingUniques(UniqueType.GreatPersonPointPercentage))
+            if (cityInfo.matchesFilter(unique.params[1]))
+                cityInfo.currentGPPBonus += unique.params[0].toInt()
+        // Sweden UP
+        for (otherCiv in cityInfo.civInfo.getKnownCivs()) {
+            if (!cityInfo.civInfo.getDiplomacyManager(otherCiv).hasFlag(DiplomacyFlags.DeclarationOfFriendship))
+                continue
+
+            for (ourUnique in cityInfo.civInfo.getMatchingUniques(UniqueType.GreatPersonBoostWithFriendship))
+                cityInfo.currentGPPBonus += ourUnique.params[0].toInt()
+            for (theirUnique in otherCiv.getMatchingUniques(UniqueType.GreatPersonBoostWithFriendship))
+                cityInfo.currentGPPBonus += theirUnique.params[0].toInt()
+        }
+        var specialistFoodBonus = 0f
+        for (unique in cityInfo.getMatchingUniques(UniqueType.FoodConsumptionBySpecialists))
+            if (cityInfo.matchesFilter(unique.params[1]))
+                specialistFoodBonus = -(unique.params[0].toFloat() / 100f) * 2f // base 2 food per Pop, and need to inverse negative
```

Comment:
We set this each time instead of it adding together?

Reply:
Good catch. Should be cumulative multiplication, same as in CityStats. Fixed